workflow/rules/assembly.smk

```diff
@@ -14,12 +14,59 @@ rule assembly:
     conda:
         "../envs/megahit.yaml"
     shell:
-        "(megahit -1 {input.fastq1} -2 {input.fastq2} --out-dir {params.outdir} -f && mv {params.outdir}/final.contigs.fa {output.contigs} ) 2> {log}"
+        "(zcat {input.fastq1} |wc -l > {output.read_count} && "
+        "megahit -1 {input.fastq1} -2 {input.fastq2} --out-dir {params.outdir} -f && "
+        "mv {params.outdir}/final.contigs.fa {output.contigs} ) 2> {log}"
+
+
+rule assembly_metaspades:
+    input:
+        fastq1="results/{date}/clipped-reads/{sample}.1.fastq.gz",
+        fastq2="results/{date}/clipped-reads/{sample}.2.fastq.gz",
+    output:
+        contigs="results/{date}/assembly/metaspades/{sample}/contigs.fasta",
+        read_count=temp("results/{date}/assembly/metaspades/{sample}.log"),
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    log:
+        "logs/{date}/metaSPAdes/{sample}.log",
+    threads: 16
+    conda:
+        "../envs/spades.yaml"
+    shell:
+        "(zcat {input.fastq1} |wc -l > {output.read_count} && "
+        "metaspades.py -1 {input.fastq1} -2 {input.fastq2} -o {params.outdir} -t {threads}) > {log} 2>&1"
+
+
+rule mv_contigs:
+    input:
+        contigs=get_contigs,
+    output:
+        contigs=temp("results/{date}/assembly/{sample}.contigs.fasta"),
+    log:
+        "logs/{date}/mv_contigs/{sample}.log",
+    conda:
+        "../envs/unix.yaml"
+    shell:
+        "(cp {input.contigs} {output.contigs}) > {log} 2>&1"
+
+
+rule mv_assembly_read_counts:
+    input:
+        read_counts=get_read_counts,
+    output:
+        read_counts=temp("results/{date}/assembly/{sample}.log"),
+    log:
+        "logs/{date}/mv_read_counts/{sample}.log",
+    conda:
+        "../envs/unix.yaml"
+    shell:
+        "(cp {input.read_counts} {output.read_counts}) > {log} 2>&1"
```

Comment:
These two rules seem unnecessary to me. What is the point?

Reply:
edited assemblies and also functions in commons.smk